### PR TITLE
[dagster-io/ui] Move fonts to ui, use base64 encoding

### LINKED ISSUE #6250: Dagit fonts fail to load when using --path-prefix
## Summary
I see errors 404 on `.woff|.woff2` files when using `--path-prefix` to run dagit. All other links within dagit are ok.

## Reproduction
I am running dagster 0.13.14 in docker with the following command (copied from my `docker-compose.yml`):
```yaml
      - dagit
      - -h
      - "0.0.0.0"
      - "--path-prefix"
      - "/dagit"
      - -p
      - "3000"
      - -w
      - workspace.yaml
```

I can confirm the path prefix is working fine as all other assets (svg, js, css...) are found as expected. The fonts are correctly served: if I replace the problematic url given by dagit like:

`https://example.com/static/media/inconsolata-v21-latin-regular.5defbb9e.woff2`

with
`https://example.com/dagit/static/media/inconsolata-v21-latin-regular.5defbb9e.woff2`

I can download the font files.

## Additional Info about Your Environment

This seems to be a regression, as #3777 seems to have fixed this issue a few months ago already.

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by hellendag**
## Summary

Resolves #6250.

This PR does a handful of things associated with our fonts:

- Move font CSS into `@dasgter-io/ui` instead of `dagit-core`, so that fonts are present for all future consumers of the library.
- Use Base64 encoding instead of woff2 files for the fonts themselves.
- Clean up CSS imports on Storybook roots.

The Base64 encoding is the fix for the bug in the attached issue. The problem is that the webpack build uses absolute paths for the `url()` values in the font CSS, which cannot be made to match a runtime `--path-prefix` value for Dagit. (The path prefix is inserted into the built html at runtime, but we don't have access to these `url()` paths at that point.) To resolve the problem of the font files being unavailable, we can use Base64 encoding to include them directly within the CSS.

This necessitates adding `data:` to the `font-src` CSP entry. We'll also need to make a corresponding change in dagit-cloud, and I added a comment to that effect in `fonts.css`.

The Base64 itself isn't too heavy overall, so I'm not really worried about any major impact on build size here.

## Test Plan

Run Dagit with `--path-prefix=/foo`, verify that all fonts load and render properly from Base64.

Tested in Chrome, Firefox, Safari, to sanity check font loading/rendering.

View Storybook for `dagit-core` and `ui`, verify same. Verify also that Blueprint CSS is behaving correctly, as this is now coming through the `ui` library.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6253**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: js_modules/dagit/packages/core/src/app/AppProvider.tsx [hunk 1]
```diff
@@ -1,5 +1,3 @@
-import '../fonts/fonts.css';
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/AppProvider.tsx:1]: This import is now implicit by making use of the `@dagster-io/ui` library.

**Review by bengotow — APPROVED** [review]
Hmm, I'm surprised the issue with the custom dagit path prefix hadn't come up already - this seems like a good solution to me.  👍

**Comment by hellendag** [comment]
> Hmm, I'm surprised the issue with the custom dagit path prefix hadn't come up already - this seems like a good solution to me. 👍

It came up a while back but I think the problem went away when we switched to system fonts.

Then, when we brought in Inter and Inconsolata, it was fine while we were using the Google Fonts API. I had to switch back to local fonts because offline mode didn't work with the Fonts API, which brought us full circle to the font files failing to load on path-prefix. I'm just surprised the bug didn't get reported sooner!

Hopefully this resolves both offline mode and path-prefix, permanently.

**Pull request merged**